src/core/hooks/useDynamicRelativePosition.tsx

```diff
@@ -0,0 +1,37 @@
+import { useEffect, useRef, useState } from 'react';
+
+export const useRelativePositioning = (title: string, characterLimit: number) => {
+  const containerRef = useRef<HTMLDivElement | null>(null);
+  const titleRef = useRef<HTMLDivElement | null>(null);
+  const filterRef = useRef<HTMLDivElement | null>(null);
+  const [shouldPositionBelow, setShouldPositionBelow] = useState(false);
+  const isShortTitle = title.length < characterLimit;
+
+  useEffect(() => {
+    const updatePosition = () => {
+      if (containerRef.current && titleRef.current && filterRef.current) {
+        const containerWidth = containerRef.current.offsetWidth;
+        const titleWidth = titleRef.current.offsetWidth;
+        const filterWidth = filterRef.current.offsetWidth;
+
+        setShouldPositionBelow(titleWidth + filterWidth > containerWidth);
+      }
+    };
+
+    updatePosition();
+    window.addEventListener('resize', updatePosition);
```

Comment:
Avoid this if possible, better:

import { useWindowWidth } from '@react-hook/window-size';

then in component:
const width = useWindowWidth();

useEffect(() => {
    const updatePosition = () => {
      if (containerRef.current && titleRef.current && filterRef.current) {
        const containerWidth = containerRef.current.offsetWidth;
        const titleWidth = titleRef.current.offsetWidth;
        const filterWidth = filterRef.current.offsetWidth;

        setShouldPositionBelow(titleWidth + filterWidth > containerWidth);
      }
    };
    updatePosition();
  }, [title, width]);

Reply:
This hook: @react-hook/window-size handles resizing in the best possible way.

Reply:
Thanks for the feedback, done